spock-specs/src/test/groovy/org/spockframework/mock/AdditionalInterfaceResponseSpec.groovy

```diff
@@ -0,0 +1,68 @@
+package org.spockframework.mock
+
+import spock.lang.Issue
+import spock.lang.Specification
+
+import java.util.function.IntSupplier
+
+class AdditionalInterfaceResponseSpec extends Specification {
+
+  @Issue("https://github.com/spockframework/spock/issues/1405")
+  def "Defining responses for additionalInterfaces for Groovy class Ticket #1405"() {
+    setup:
+    def a = Stub(A, additionalInterfaces: [B]) {
```

Comment:
I have added two test cases for that.
But it is not that great, becaus the cast to the additional interface is failing, because we do not spawn a separate class for that. I have made a note in the test cases for that.